executor/builder.go

```diff
@@ -1329,6 +1332,10 @@ func (b *executorBuilder) buildSelectInto(v *plannercore.SelectInto) Executor {
 }
 
 func (b *executorBuilder) buildUnionScanExec(v *plannercore.PhysicalUnionScan) Executor {
+	b.encounterUnionScan = true
```

Comment:
not sure
unionScan may be embedded with unionScan, we should restore the old value here

Reply:
Done